### PR TITLE
Generate stable bindata output

### LINKED ISSUE #30621: hack/update-bindata.sh sets KUBE_ROOT default value relative to test/e2e/generated/
I tried to launch hack/update-bindata.sh from root directory of kubernetes repo and I got following error:

`hack/update-bindata.sh: line 26: ../../..//cluster/lib/logging.sh: No such file or directory`

After looking the code, I understood that update-bindata.sh script should be launched only from test/e2e/generated/ directory or I should specify KUBE_ROOT variable by myself. It is not user-friendly at all and it would be nice to align all pathes to `${BASH_SOURCE[0]}`, so users will be able to launch this script from whatever they want path.

### PR DESCRIPTION
**Pull request by liggitt**
fixes #30621

bindata generation produces spurious diffs when run with different go versions because gzip output is not stable between versions.

It also produces spurious diffs when invoked directly vs via make/go:generate.

This PR:
* adds data to bindata uncompressed (makes translation files human-readable, and the zip files were already compressed... for reference the file was 27.7 KB before, and 20.1 KB with this change)
* runs the generation from KUBE_ROOT, so it doesn't matter whether it is invoked directly or via go:generate

supercedes https://github.com/kubernetes/kubernetes/pull/39631

### CODE REVIEW

**Comment by deads2k** [comment]
This worked for me.

File: hack/generate-bindata.sh [hunk 1]
```diff
@@ -63,7 +63,7 @@ rm -f "${BINDATA_OUTPUT}.tmp"
 
 # These are files for runtime code
 BINDATA_OUTPUT="${KUBE_ROOT}/pkg/generated/bindata.go"
-go-bindata -nometadata -prefix "${KUBE_ROOT}" -o "${BINDATA_OUTPUT}.tmp" -pkg generated \
+go-bindata -nometadata -nocompress -prefix "${KUBE_ROOT}" -o "${BINDATA_OUTPUT}.tmp" -pkg generated \
```
Comment (deads2k) [inline hack/generate-bindata.sh]: You ok doing this?  It makes my life a lot easier, so I can definitely live with it

Comment (liggitt) [inline hack/generate-bindata.sh]: I prefer it, actually... half the things being added are useful if they are human readable, and the other half are already zip-compressed

**Comment by liggitt** [comment]
fixed the difference between direct invocation and invocation via go:generate by doing pushd/popd and running the guts of the bindata generation from KUBE_ROOT

**Comment by liggitt** [comment]
openapi flake fixed.
@k8s-bot unit test this

**Comment by fabianofranz** [comment]
This looks better with `-nocompress` than without it. LGTM but I'll defer to @brendandburns.

**Comment by liggitt** [comment]
rebased

**Pull request merged**